src/algebraic_geometry/Spec.lean

```diff
@@ -211,5 +211,30 @@ Spec, as a contravariant functor from commutative rings to locally ringed spaces
   map_id' := λ R, by rw [unop_id, Spec.LocallyRingedSpace_map_id],
   map_comp' := λ R S T f g, by rw [unop_comp, Spec.LocallyRingedSpace_map_comp] }
 
+section Spec_Γ
+open algebraic_geometry.LocallyRingedSpace
+
+/-- The morphism `R ⟶ Γ(Spec R)` given by `algebraic_geometry.structure_sheaf.to_open`.  -/
+def to_Spec_Γ (R : CommRing) : R ⟶ Γ.obj (op (Spec.to_LocallyRingedSpace.obj (op R)))
+  := structure_sheaf.to_open R ⊤
+
+instance is_iso_to_Spec_Γ (R : CommRing) : is_iso (to_Spec_Γ R)
+  := by convert is_iso_to_global R; cases R; refl
+
+lemma Spec_Γ_naturality {R S : CommRing} (f : R ⟶ S)
+  : f ≫ to_Spec_Γ S = to_Spec_Γ R ≫ Γ.map (Spec.to_LocallyRingedSpace.map f.op).op
+:= by ext x p; symmetry; apply localization.local_ring_hom_to_map
+
+/-- The counit of the adjunction `Γ ⊣ Spec` is an isomorphism. -/
+def Spec_Γ_identity : Spec.to_LocallyRingedSpace.right_op ⋙ Γ ≅ 𝟭 _ := by {
+  symmetry,
+  apply nat_iso.of_components,
+  swap, intro R,
+    convert global_sections_iso R; cases R; dsimp; refl,
+  intros R S f,
+  convert Spec_Γ_naturality f; cases R; cases S; dsimp; refl
```

Comment:
After applying `convert Spec_Γ_naturality f`, you get two goals of the form `(global_sections_iso R).hom = to_Spec_Γ R`. This could be a lemma, in which case you could `erw` before `convert Spec_Γ_naturality f`.